src/Analyzers/MSTest.GlobalConfigsGenerator/AnalyzerSeverityDecider.cs

```diff
@@ -0,0 +1,79 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using Microsoft.CodeAnalysis;
+
+// NOTE for our current matrix.
+// We decide whether to include a rule in a specific mode or not based on two factors.
+// 1. IsEnabledByDefault (true/false)
+// 2. DefaultSeverity (Info, Warn) - note: we currently don't use severity Hidden or Error.
+// So, we have 4 possible combinations.
+// 1. IsEnabledByDefault = true, DefaultSeverity = Warn (we include that in All, Recommended, and Default)
+// 2. IsEnabledByDefault = true, DefaultSeverity = Info (we include that in All and Recommended)
+// 3. IsEnabledByDefault = false, DefaultSeverity = Warn (we don't combine Warn with not enabled by default)
+// 4. IsEnabledByDefault = false, DefaultSeverity = Info (we include that in All only)
+internal static class AnalyzerSeverityDecider
+{
+    public static DiagnosticSeverity? GetSeverity(DiagnosticDescriptor rule, MSTestAnalysisMode mode)
+    {
+        DiagnosticSeverity? severity = mode switch
+        {
+            MSTestAnalysisMode.All => DecideForModeAll(rule),
+            MSTestAnalysisMode.Recommended => DecideForModeRecommended(rule),
+            MSTestAnalysisMode.Default => DecideForModeDefault(rule),
+            MSTestAnalysisMode.None => DecideForModeNone(rule),
+            _ => throw new ArgumentException($"Unexpected MSTestAnalysisMode '{mode}'.", nameof(mode)),
+        };
+
+        return severity;
+    }
+
+    private static DiagnosticSeverity? DecideForModeAll(DiagnosticDescriptor rule) =>
+        // TODO: We should consider at least not enabling "conflicting" rules.
+        // Or alternatively, have a configuration with reasonable default for such rules.
+        // Example of conflicting rules is an analyzer that suggests Dispose instead of
+        // cleanup, and another analyzer that suggests the opposite.
```

Comment:
For this example, I am wondering if we should have either some user defined property/editorconfig or have more profiles. We have a few rules that are more around "favor .NET lifecycle" (in which case we prefer ctor and dispose) or favor consistency in which case we go for the opposite.

Reply:
We could go with an editorconfig option to determine the preferred style for "conflicting" rules and have what we think is a "reasonable" default. Then the analyzer can get the property value and decide if it should analyzer or not. i.e, it will be that only one of the two can be analyzing. It's kind of a separate feature but is related so I can have at least a quick investigation on that.